Reduce usage of the platform dependent `unsigned int` type

hebasto opened the pull request:
`unsigned int` is a platform dependent type.

hebasto committed:
Replace platform dependent type with proper const

hebasto commented:
@achow101 Would you mind to look into [the Travis build log](https://travis-ci.org/bitcoin/bitcoin/jobs/440074328) to find out the reason of the `rpc_psbt.py ` functional test failure?
Does 54a546f brake the code in any way?

practicalswift commented:
@hebasto That was likely just a temporary Travis hiccup. I've now restarted the build job :-)

hebasto commented:
@practicalswift 

> I've now restarted the build job :-)

Thank you!

hebasto commented:
I've investigated in direction pointed by @practicalswift and found two cases which are, IMO, safe to fix (i.e. not belong to the consensus part or other critical parts of the code):

1) this LOC
https://github.com/bitcoin/bitcoin/blob/f34c8c466a0e514edac2e8683127b4176ad5d321/src/test/skiplist_tests.cpp#L173
effectively duplicates that one:
https://github.com/bitcoin/bitcoin/blob/f34c8c466a0e514edac2e8683127b4176ad5d321/src/test/skiplist_tests.cpp#L169

2) another case in the `src/bloom.cpp`.

hebasto commented:
@practicalswift 
Thank you for your review and a piece of advice.

> Perhaps worth investigating if you can find any similar cases in:
> `$ git grep -E 'std::numeric_limits<.*>::(min|max)' | grep -vE 'int[0-9]'`

It's done. Please re-review.

practicalswift commented on src/test/skiplist_tests.cpp:
```diff
@@ -170,7 +170,6 @@ BOOST_AUTO_TEST_CASE(findearliestatleast_edge_test)
     BOOST_CHECK_EQUAL(chain.FindEarliestAtLeast(-1)->nHeight, 0);
 
     BOOST_CHECK_EQUAL(chain.FindEarliestAtLeast(std::numeric_limits<int64_t>::min())->nHeight, 0);
-    BOOST_CHECK_EQUAL(chain.FindEarliestAtLeast(std::numeric_limits<unsigned int>::min())->nHeight, 0);
```
How is this one redundant? :-)

hebasto replied:
`std::numeric_limits<unsigned int>::min() == 0`, therefore, this LOC duplicates that one:
https://github.com/bitcoin/bitcoin/blob/f34c8c466a0e514edac2e8683127b4176ad5d321/src/test/skiplist_tests.cpp#L169

practicalswift replied:
Ah saw it now! Thanks

hebasto commented on src/bloom.cpp:
```diff
@@ -301,7 +301,7 @@ bool CRollingBloomFilter::contains(const uint256& hash) const
 void CRollingBloomFilter::reset()
 {
-    nTweak = GetRand(std::numeric_limits<unsigned int>::max());
+    nTweak = GetRand(std::numeric_limits<uint64_t>::max());
```
https://github.com/bitcoin/bitcoin/blob/eb7daf4d600eeb631427c018a984a77a34aca66e/src/random.h#L23

gmaxwell commented on src/policy/rbf.cpp:
```diff
@@ -7,7 +7,7 @@
 {
     for (const CTxIn &txin : tx.vin) {
-        if (txin.nSequence < std::numeric_limits<unsigned int>::max()-1) {
+        if (txin.nSequence < 0xffffffff - 1) {
```
<= MAX_BIP125_RBF_SEQUENCE

hebasto committed:
Remove duplicated code

The deleted LOC is a dup so far as
`std::numeric_limits<unsigned int>::min()` == 0

hebasto commented:
@gmaxwell 
Thank you for your review.
Fixed. Please re-review.

sipa commented:
I believe `unsigned int` is 32 bits on all our supported platform.

ACK on the change, but there must be many more places where the size of a unsigned int is assumed?

hebasto commented:
@sipa 
Thank you for your review.

> ... there must be many more places where the size of a unsigned int is assumed?

Yes. I've found some dozens of them :)
But those cases belong to the consensus part or other critical parts of the code which I'd prefer to not touch.

MarcoFalke commented:
utACK 917590980be69f83291613e3b8ed6d14a1b30619

fanquake commented:
I've asked @ryanofsky to pull your changes into #14636, as that PR has more extensive changes, so I'll close this for now.
